homeassistant/components/overkiz/config_flow.py

```diff
@@ -132,14 +214,96 @@ async def async_step_user(
                 )
 
         return self.async_show_form(
-            step_id="user",
+            step_id="cloud",
             data_schema=vol.Schema(
                 {
-                    vol.Required(CONF_USERNAME, default=self._default_user): str,
+                    vol.Required(CONF_USERNAME, default=self._user): str,
                     vol.Required(CONF_PASSWORD): str,
-                    vol.Required(CONF_HUB, default=self._default_hub): vol.In(
-                        {key: hub.name for key, hub in SUPPORTED_SERVERS.items()}
-                    ),
+                }
+            ),
+            description_placeholders=description_placeholders,
+            errors=errors,
+        )
+
+    async def async_step_local(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the local authentication step via config flow."""
+        errors = {}
+        description_placeholders = {}
+
+        if user_input:
+            self._host = user_input[CONF_HOST]
+            self._user = user_input[CONF_USERNAME]
+
+            # inherit the server from previous step
+            user_input[CONF_HUB] = self._server
+
+            try:
+                user_input = await self.async_validate_input(user_input)
+            except TooManyRequestsException:
+                errors["base"] = "too_many_requests"
+            except BadCredentialsException:
+                errors["base"] = "invalid_auth"
+            except ClientConnectorCertificateError as exception:
+                errors["base"] = "certificate_verify_failed"
+                LOGGER.debug(exception)
+            except (TimeoutError, ClientError) as exception:
+                errors["base"] = "cannot_connect"
+                LOGGER.debug(exception)
+            except MaintenanceException:
+                errors["base"] = "server_in_maintenance"
+            except TooManyAttemptsBannedException:
+                errors["base"] = "too_many_attempts"
+            except NotSuchTokenException:
+                errors["base"] = "not_such_token"
+            except DeveloperModeDisabled:
+                errors["base"] = "developer_mode_disabled"
+            except UnknownUserException:
+                # Somfy Protect accounts are not supported since they don't use
+                # the Overkiz API server. Login will return unknown user.
+                description_placeholders["unsupported_device"] = "Somfy Protect"
+                errors["base"] = "unsupported_hardware"
+            except Exception as exception:  # pylint: disable=broad-except
+                errors["base"] = "unknown"
+                LOGGER.exception(exception)
```

Comment:
I think `.exception()` automatically captures the current exception context. The first positional argument is a message to log with the exception, not the exception itself?

Reply:
Will have a look, this is currently how it is configured in the Config Flow already, so perhaps we need to change that as well.

Reply:
You are right, fixed! Thanks @MarijnS95